### ISSUE TITLE
Multiple levels of catch-all routes with intercepted + parallel routes loads the wrong route on intercept, but the correct route on initial page load

### ISSUE THREAD

**Issue by augustl** [vercel/next.js#58660]
### Link to the code that reproduces this issue

https://github.com/augustl/nextjs-intercepting-modal-bug

### To Reproduce

1. Open the home page at localhost:3000/
2. Click either "Show thing 1" or "Show thing 2"
3. Observe the error 

### Current vs. Expected behavior

I expected the click of "Show thing 1" or "Show thing 2" to load the intercepted route, `src/app/@modal/(.)things/[...ids]/page.tsx`. Instead, it attempts to load the top-level catch all route `src/app/[...slug]/page.tsx`, as evident by the error message that refers to a line in that file caused by parameters being null due to the route loading mismatch.

Also observe that when refreshing the page to go directly to localhost:3000/things/1, the page loads correctly and without error.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
Binaries:
  Node: 18.18.2
  npm: 9.8.1
  Yarn: 1.22.21
  pnpm: N/A
Relevant Packages:
  next: 14.0.4-canary.4
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Routing (next/router, next/navigation, next/link)

### Additional context

The combination of a top level catch-all route and a catch-all segment in the `/things/[...ids]` route are both needed to trigger this issue. If you remove the top level catch-all `[...slug]`, or change `/things/[...ids]` to something like `/things/[id]`, it works fine.

<sub>[NEXT-1809](https://linear.app/vercel/issue/NEXT-1809/multiple-levels-of-catch-all-routes-with-intercepted-parallel-routes)</sub>

**Comment by augustl** [comment]
Potentially related to #51784, #58272 and #48719

**Comment by augustl** [comment]
Fix works great (14.0.5-canary.24), thanks 🥳
